### PR TITLE
Fix: #3. Add different sections for YAML and OAS.

### LINKED ISSUE #3: numbered sections for each type
it's maybe just OCD (and because it's very useful for webconcepts.info to have linkable references) but i think it's nice to have numbered sections for everything that's being defined in the spec. in this case, i suggest to have a numbered section for each of the four media types, and to have a brief explanation of what it identifies. that way, each definition is self-contained and can be easily referenced when necessary.

@ioggstream
> we could have 4 sections like this one https://httpwg.org/http-core/draft-ietf-httpbis-semantics-latest.html#multipart.byteranges

@dret
> i think this would be perfect.

### PR DESCRIPTION
**Pull request by ioggstream**
## This PR

- introduce numbered sections for YAML
- fixes #3 

@dret since each media type is defined in its IANA consideration, would it be ok for you if we just add a section for YAML?
Feel free to PR with another proposal.

### CODE REVIEW

**Comment by ioggstream** [comment]
@dret could you PTAL ? :)

File: draft-ietf-httpapi-rest-api-mediatypes.md [hunk 1]
```diff
@@ -87,12 +87,28 @@ and `application/openapi+yaml`.
...
+This specification defines two media types:
+
+- `text/yaml` when a yaml document is intended to be consumed by a
+  human reader;
+- `application/yaml` when a yaml document is intended to be consumed by
+  a machine.
+
+## The OpenAPI Specification Media Types
 
 The OpenAPI Specification Media Types convey OpenAPI Specification (OAS) files
```
Comment (darrelmiller) [inline draft-ietf-httpapi-rest-api-mediatypes.md:103]:
```suggestion
The OpenAPI Specification Media Types convey OpenAPI document (OAS) files
```
Instances of an API description are referred to as an OpenAPI document and not an "OpenAPI specification".

File: draft-ietf-httpapi-rest-api-mediatypes.md [hunk 2]
```diff
@@ -87,12 +87,28 @@ and `application/openapi+yaml`.
...
 as defined in [oas] for version 3.0.0 and above.
 
 Those files can be serialized in {{JSON}} or [yaml].
+This specification defines two media types for OpenAPI Specification files:
```
Comment (darrelmiller) [inline draft-ietf-httpapi-rest-api-mediatypes.md:107]:
```suggestion
This specification defines two media types for OpenAPI document files:
```

File: draft-ietf-httpapi-rest-api-mediatypes.md [hunk 3]
```diff
@@ -87,12 +87,28 @@ and `application/openapi+yaml`.
...
+
+- `application/openapi+json` for the JSON format;
+- `application/openapi+yaml` for the YAML format.
+
 Since there are multiple OpenAPI Specification Specifications versions,
```
Comment (darrelmiller) [inline draft-ietf-httpapi-rest-api-mediatypes.md:112]:
```suggestion
Since there are multiple OpenAPI Specification versions,
```

**Comment by dret** [comment]
On 2022-01-14 09:12, Roberto Polli wrote:
> @dret <https://github.com/dret> could you PTAL ? :)

to be honest, i prefer definitions to have URIs and not just 
registrations. but if we have URIs, that's already a lot better than not 
having URIs! thanks!

**Pull request merged**